### PR TITLE
Add an enum field to echo request and response

### PR DESCRIPTION
**Pull request by software-dov**
Implementation of #142 

Generators running integration tests against showcase may want to check enum behavior.
This change adds an enum type and field to EchoRequest and EchoResponse

### CODE REVIEW

File: schema/google/showcase/v1beta1/echo.proto [hunk 1]
```diff
@@ -106,9 +106,19 @@ service Echo {
   };
 }
 
-// The request message used for the Echo, Collect and Chat methods. If content
-// is set in this message then the request will succeed. If status is set in 
-// this message then the status will be returned as an error.
+// A throwaway enum used to test enum capabilities in GAPIC surfaces
+enum Throwaway {
```
Comment (noahdietz) [inline schema/google/showcase/v1beta1/echo.proto:110]: Can we think of a better name than `Throwaway`? To make things fit in better and form some semblance of a "real" API (I realize it's not).

Alternatively, `identity.proto` has more "context" built into it that might provide a better spot to implement an enum.

Comment (software-dov) [reply schema/google/showcase/v1beta1/echo.proto:110]: Agreed, 'Throwaway' was my throwaway name idea :P
My initial thought was that a quick enum check fit more in with echo than identity, but I'll take another look through identity and reevaluate my opinion.

Comment (software-dov) [reply schema/google/showcase/v1beta1/echo.proto]: Went with 'severity' instead of 'throwaway'

File: schema/google/showcase/v1beta1/echo.proto [hunk 2]
```diff
@@ -117,12 +127,18 @@ message EchoRequest {
     // The error to be thrown by the server.
     google.rpc.Status error = 2;
   }
+
+  // The option to be echoed by the server.
+  Severity opt = 3;
```
Comment (noahdietz) [inline schema/google/showcase/v1beta1/echo.proto:132]: Please rename the field to `severity` to match the type name.

File: schema/google/showcase/v1beta1/echo.proto (hunk 1 above)
Comment (noahdietz) [reply schema/google/showcase/v1beta1/echo.proto]: Thanks, sgtm

File: schema/google/showcase/v1beta1/echo.proto (hunk 2 above)
Comment (noahdietz) [reply schema/google/showcase/v1beta1/echo.proto:132]: And please update the leading comment too :)

File: schema/google/showcase/v1beta1/echo.proto [hunk 3]
```diff
@@ -117,12 +127,18 @@ message EchoRequest {
...
+  // The severity to be echoed by the server.
+  Severity severity = 3;
 }
 
 // The response message for the Echo methods.
 message EchoResponse {
   // The content specified in the request.
   string content = 1;
+
+  // The option specified in the request.
```
Comment (noahdietz) [inline schema/google/showcase/v1beta1/echo.proto:140]:
```suggestion
  // The severity specified in the request.
```

**Comment by noahdietz** [comment]
I think the "binary file" is some weird character added in `echo_service.go`. I checked out your fork and attempted to test things and saw the following:
Screen Shot 2020-05-12 at 5 34 55 PM

**Comment by noahdietz** [comment]
Here is what I was seeing:
Screen Shot 2020-05-12 at 5 37 18 PM

**Comment by chrisdunelm** [comment]
LGTM once the `server/services/echo_service.go` binary file issue is resolved; and @noahdietz comment suggestion is included.

**Comment by noahdietz** [comment]
Hey @software-dov sorry for the churn. With other folks contributing to gapic-showcase, Mira and I put together these steps to set up the development environment: https://github.com/googleapis/gapic-showcase#development-environment

Specifically, the section [Making changes to the protos](https://github.com/googleapis/gapic-showcase#making-changes-to-the-protos) is useful in this case. CI is failing because it regenerates all of the code from the protos and what you have checked in here for `echo.pb.go` doesn't match that, which means your tests will fail against the correctly generated code in CI. 

Can you follow the steps to setup the environment and then run `go run ./util/cmd/compile_protos`?

File: server/services/echo_service.go [hunk 4]
```diff
@@ -46,7 +46,7 @@ func (s *echoServerImpl) Echo(ctx context.Context, in *pb.EchoRequest) (*pb.Echo
 		return nil, err
 	}
 	echoTrailers(ctx)
-	return &pb.EchoResponse{Content: in.GetContent()}, nil
+	return &pb.EchoResponse{Content: in.GetContent(), Severity: in.Severity}, nil
```
Comment (noahdietz) [inline server/services/echo_service.go:49]: Please use the Getter for `Severity`: `Severity: in.GetSeverity()`

Comment (software-dov) [reply server/services/echo_service.go:49]: Added a getter and started using it.

File: server/services/echo_service_test.go [hunk 5]
```diff
@@ -39,7 +39,10 @@ func TestEcho_success(t *testing.T) {
 
 	server := NewEchoServer()
 	for _, val := range table {
-		in := &pb.EchoRequest{Response: &pb.EchoRequest_Content{Content: val}}
+		in := &pb.EchoRequest{
+			Response: &pb.EchoRequest_Content{Content: val},
+			Severity: pb.Echo_CRITICAL,
```
Comment (noahdietz) [inline server/services/echo_service_test.go:44]: This is incorrect, the regenerated code creates an enum named `Severity_CRITICAL`. Please follow the instructions for setting up the [generation environment](https://github.com/googleapis/gapic-showcase#development-environment). Then, please regenerate the protobuf/gRPC package via `go run ./util/cmd/compile_protos`.

**Pull request merged**